connection-api/src/main/java/org/terracotta/connection/ConnectionService.java

```diff
@@ -52,4 +65,15 @@
    * @throws ConnectionException on connection failure
    */
   Connection connect(URI uri, Properties properties) throws ConnectionException;
+
+  /**
+   * Establish a connection to the given list of servers using the specified properties. handlesScheme() must be called
+   * with the scheme prior to connect(). Calling connect() on an unverified scheme can lead to unspecified behavior.
+   *
+   * @param servers list of servers to connect to
+   * @param properties user specified implementation specific properties
+   * @return established connection
+   * @throws ConnectionException on connection failure
+   */
+  Connection connect(List<String> servers, Properties properties) throws ConnectionException;
```

Comment:
Can this be `connect(String scheme, Iterable<InetSocketAddress> listOfServers, Properties cps)` or `connect(SomeEnumType scheme, Iterable<InetSocketAddress> listOfServers, Properties cps)`?

Reply:
Should `ConnectionFactory.connect` have a similar signature too then?
1. If yes, the upper layers will have to directly pass `Iterable<InetSocketAddress> listOfServers`. Seems somewhat inconvenient, but would save us the validation effort.
2. If no, should it accept a `List<String> servers` and parse the `String`s into `InetSocketAddress`es?